### DISCUSSION TITLE
Is there a way to setup the 'ToStringTag' for c# class?

### DISCUSSION THREAD

**Discussion by fantasydr** [sebastienros/jint#1649] [Q&A] [answered]
Hi,

I've tried to use TypeReference.CreateTypeReference to define the class in js for my c# class.
```
_engine.SetValue("MyClass", TypeReference.CreateTypeReference(_engine, typeof(MyClass)));
```

The problem is that if I created a instance of MyClass in javascript, 
```
var myclass = new MyClass();
Object.prototype.toString.call(myclass)
```
It always return '[object Object]'

I tried to define the toStringTag in js, it works as expected.
```
class MyClassInJs {
  get [Symbol.toStringTag]() {
    return 'MyClassInJs';
  }
}
```

But how could I do the same thing for the class defined in c#?
I already tried to set the property on TypeReference, it won't work.
```
var type = typeof(MyClass);
var reference = TypeReference.CreateTypeReference(_engine, type);
reference.FastSetProperty(GlobalSymbolRegistry.ToStringTag, new PropertyDescriptor(type.Name, false, false, true));
```

Thanks a lot for the help!

**Comment by lahma** [comment, marked as answer]
Jint seems to have some trouble understanding the reading of property from wrapped object's type reference. You can work around it by having a custom object wrapper logic at least.

```c#
var engine = new Engine(options => options.SetWrapObjectHandler((e, target, t) =>
{
    var wrapper = new ObjectWrapper(e, target, t);
    wrapper.FastSetProperty(GlobalSymbolRegistry.ToStringTag, new PropertyDescriptor(target.GetType().Name, false, false, true));
    return wrapper;
}));
```

**Reply by fantasydr** [reply]
Thank you for the quick answer!
I am totally lost with the Jint interfaces and can't find a proper way to define the prototype...

**Reply by lahma** [reply]
This has now been fixed to be supported, the test case showing the usage passes in `main` (available from the MyGet feed).

https://github.com/sebastienros/jint/blob/f12cb7d4cd973065fb9fb6c4540255bffc34a13b/Jint.Tests/Runtime/InteropTests.TypeReference.cs#L186-L198

**Reply by fantasydr** [reply]
Thank you for the fix! I wanted to confirm if this fix has been included in the '3.0.0-beta-2053' version?
When I tried using Nuget to fetch Jint, it appears unchanged; I'm still unable to access the ToStringTag property.
However, when I reference the source code repository directly, your fix works perfectly.

There's an edge case that may impact the fix:
When an object is created directly within JavaScript, everything functions as expected. However, if the object is initiated in my C# code and then returned to JavaScript, the toString.call(c) still yields 'object Object'.

```
var c = createClassInCSharp();
console.log(Object.prototype.toString.call(c)); // didn't change, still [object Object]

c = new MyClass();
console.log(Object.prototype.toString.call(c)); // works fine
```

The workaround employing 'SetWrapObjectHandler' is effective in both scenarios.

**Reply by lahma** [reply]
Like said, only on MyGet feed for now and later on NuGet as part of next release. Need to check that edge at some point...

**Reply by lahma** [reply]
Above should now also work and available soon on MyGet feed.
